cli/azd/pkg/tools/dotnet/dotnet.go

```diff
@@ -14,8 +14,9 @@ import (
 
 type DotNetCli interface {
 	tools.ExternalTool
-	Publish(ctx context.Context, project string, output string) error
 	Restore(ctx context.Context, project string) error
+	Build(ctx context.Context, project string, output string) error
```

Comment:
I think we should be using `dotnet publish` instead of `dotnet build` here, and if you agree, I think we can remove this.  If we are keeping it, we should think about if we still need `Publish` on this interface, since I think it is now dead code?